taxonomy-editor-frontend/src/pages/editentry/LanguageSelectionDialog.tsx

```diff
@@ -52,15 +54,25 @@ const LanguageSelectionDialog = ({
               selected.map((langCode) => ISO6391.getName(langCode)).join(", ")
             }
           >
-            {ISO6391.getAllNames()
+            {["All languages"]
+              .concat(ISO6391.getAllNames())
               .sort()
               .map((languageNameItem) => {
-                const languageCodeItem = ISO6391.getCode(languageNameItem);
+                const languageCodeItem =
+                  languageNameItem === "All languages"
+                    ? "xx"
+                    : ISO6391.getCode(languageNameItem);
                 return languageCodeItem === mainLanguageCode ? null : (
                   <MenuItem key={languageCodeItem} value={languageCodeItem}>
```

Comment:
I feel like this `MenuItem` should also be `disabled` if the checkbox is disabled, because currently we can still click on the "All languages" option even if the checkbox looks disabled